### ISSUE TITLE
False positive triggers for `async_without_await` in `NSObjectProtocol`

### ISSUE THREAD

**Issue by akuzminskyi** [realm/SwiftLint#6422]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

I'm using the following method from [WKNavigationDelegate](https://developer.apple.com/documentation/webkit/wknavigationdelegate), and my implementation doesn't have any `await` but `async_without_await` suggests removing it. 
```swift
    func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction) async -> WKNavigationActionPolicy {
        /// .....
        return .cancel
    }
```
After that Xcode produces a warning. Looks like this happens because the protocol is `NSOblectProtocol` and requires all attributes.  I'm not sure if `SwiftSyntax` has the ability to detect it to ignore/skip it. 

<img width="690" height="171" alt="Image" src="https://github.com/user-attachments/assets/2d3c4920-4611-4233-9d2f-1b4df1440745" />

Mention the command or other SwiftLint integration method that caused the issue. Include stack traces or command output.

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version: 0.63.0
* Xcode version: 26.0.1
* Installation method used: [Universal macOS Binary](https://github.com/realm/SwiftLint/releases/download/0.63.0/portable_swiftlint.zip)
* Configuration file:

```yml
only_rules:
- async_without_await
```

Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? If so, paste their
relative paths and respective contents.

**Comment by SimplyDanny** [comment]
Unfortunately, there is no way to find out about the signature of the method in the protocol. So I fear, we just have to accept this false positive. Normally, implementing `async` protocol requirements with non-`async` variants is fine. That doesn't seem to be the case of `@objc` protocols, though.
